packages/sanity/src/core/comments/context/comments/CommentsProvider.tsx

```diff
@@ -78,12 +79,19 @@ export const CommentsProvider = memo(function CommentsProvider(props: CommentsPr
     selectedCommentId,
     isConnecting,
     onPathOpen,
+    perspective,
   } = props
   const commentsEnabled = useCommentsEnabled()
   const [status, setStatus] = useState<CommentStatus>('open')
   const {client, createAddonDataset, isCreatingDataset} = useAddonDataset()
   const publishedId = getPublishedId(documentId)
-  const editState = useEditState(publishedId, documentType, 'low')
+
+  const bundlePerspective = perspective?.startsWith('bundle.')
+    ? perspective.split('bundle.').at(1)
+    : undefined
+
+  // TODO: Allow versions to have separate comments.
+  const editState = useEditState(publishedId, documentType, 'default', bundlePerspective)
```

Comment:
I think we may want to update the `documentValue` memo below to account for the `editState.version` also.
See line 101

Reply:
This is for a follow up, but We may want also to update line 123 to use the version document id when creating a comment inside a version document. I think comments in version documents should be shown only in the version doc and not in the draft or published version

Reply:
Great points, and you're absolutely right that this (almost certainly incorrectly) does not allow users to leave different comments on different versions. Taking a look...

Reply:
I added two tickets to enable **tasks** and **comments** in version documents, we can look at that in those tickets. 
But I think we may want to do the change on line 101 on this PR, it could be harder to identify later why the comments provider is not working as expected.

Reply:
Thanks! Made that change.